client/src/state-machine.test.ts

```diff
@@ -59,31 +94,104 @@ describe("wowMachine", () => {
-          },
-        ],
-      }),
+      [NOT_REG_URLS.BUILDINGINFO_URL]: mockJsonResponse<BuildingInfoResults>(
+        SAMPLE_BUILDING_INFO_RESULTS
+      ),
     });
 
     const wm = interpret(wowMachine).start();
     wm.send(SEARCH_EVENT);
     await waitUntilStateMatches(wm, "unregisteredFound");
   });
+
+  it("should deal w/ nycha addresses", async () => {
+    mockResponses({
+      [SEARCH_URL]: mockJsonResponse(NYCHA_URLS.GEOCODING_EXAMPLE_SEARCH),
+      [NYCHA_URLS.ADDRESS_URL]: mockJsonResponse<SearchResults>({
+        addrs: [],
+        geosearch: {
+          geosupportReturnCode: "00",
+          bbl: "3004040001",
+        },
+      }),
+      [NYCHA_URLS.BUILDINGINFO_URL]: mockJsonResponse<BuildingInfoResults>(
+        SAMPLE_BUILDING_INFO_RESULTS
+      ),
+    });
+
+    const wm = interpret(wowMachine).start();
+    wm.send(SEARCH_EVENT);
+    await waitUntilStateMatches(wm, "nychaFound");
+  });
+
+  it("should deal w/ addresses with portfolios", async () => {
+    mockResponses({
+      [SEARCH_URL]: mockJsonResponse(PORTFOLIO_URLS.GEOCODING_EXAMPLE_SEARCH),
+      [PORTFOLIO_URLS.ADDRESS_URL]: mockJsonResponse<SearchResults>({
+        addrs: SAMPLE_ADDRESS_RECORDS,
+        geosearch: {
+          geosupportReturnCode: "00",
+          bbl: "3012380016",
+        },
+      }),
+    });
+
+    const wm = interpret(wowMachine).start();
+    wm.send(SEARCH_EVENT);
+    await waitUntilStateMatches(wm, "portfolioFound");
+  });
+
+  it("should deal w/ viewing timeline data", async () => {
+    mockResponses({
+      [PORTFOLIO_URLS.INDICATORS_URL]: mockJsonResponse<IndicatorsHistoryResults>(
+        SAMPLE_TIMELINE_DATA
+      ),
+    });
+
+    const wm = interpret(wowMachine).start({ portfolioFound: { timeline: "noData" } });
+    wm.state.context = PORTFOLIO_FOUND_CTX;
+
+    wm.send({ type: "VIEW_TIMELINE" });
+    await waitUntilStateMatches(wm, { portfolioFound: { timeline: "success" } });
+  });
+
+  it("should deal w/ timeline data request errors", async () => {
+    mockResponses({ [PORTFOLIO_URLS.INDICATORS_URL]: { status: 500 } });
+
+    const wm = interpret(wowMachine).start({ portfolioFound: { timeline: "noData" } });
+    wm.state.context = PORTFOLIO_FOUND_CTX;
+
+    wm.send({ type: "VIEW_TIMELINE" });
+    await waitUntilStateMatches(wm, { portfolioFound: { timeline: "error" } });
+  });
+
+  it("should deal w/ viewing summary data", async () => {
+    mockResponses({
+      [PORTFOLIO_URLS.SUMMARY_URL]: mockJsonResponse<SummaryResults>(SAMPLE_SUMMARY_DATA),
+    });
+
+    const wm = interpret(wowMachine).start({ portfolioFound: { summary: "noData" } });
+    wm.state.context = PORTFOLIO_FOUND_CTX;
+
+    wm.send({ type: "VIEW_SUMMARY" });
+    await waitUntilStateMatches(wm, { portfolioFound: { summary: "success" } });
+  });
+
+  it("should deal w/ summary data request errors", async () => {
+    mockResponses({ [PORTFOLIO_URLS.SUMMARY_URL]: { status: 500 } });
```

Comment:
Mocking a 500 status here actually triggers a console.log of the error in APIClient.ts... so while these tests pass they do clutter our testing feed with error messages like
```
HTTPError: HTTP Error Internal Server Error
        at checkStatus (/Users/sam/Sites/who-owns-what/client/src/components/APIClient.ts:149:17) {...
```

Is there any way to disable those for when we are running the tests?

Reply:
Hmm good question, honestly I think we should just remove the console.log in this case--it's not particularly helpful, at least with 500's, since the server will be logging it on its end, and with a lot more information about what caused the error?

Reply:
Yeah definitely. Ok good!